astropy/units/typing.py

```diff
@@ -11,11 +18,16 @@
 import numpy as np
 import numpy.typing as npt
 
-from astropy.units import Quantity
+from astropy.units import Quantity, UnitBase
 
 if TYPE_CHECKING:
     from typing import TypeAlias
 
+UnitLike: TypeAlias = UnitBase | str | Quantity
+"""Type alias for input that can be converted to a Unit.
```

Comment:
This should be linked to "unit-like" and vice versa. To the maximum extent that static typing allows, the two definitions should be the same.

Reply:
Yes, would certainly be good to link to `unit-like` here - I guess just use 
```
:term:`unit-like`
```
somewhere in the text? 

The other way around would seem out of scope here - we should then update the whole glossary, but that one is really for human consumption, so it needs some thought.

Reply:
Added to `UnitLike` - and this works. As said above, I don't think we should cross-link from the glossary, mostly because that should tackle all, not just one item, but also because I'm not sure it makes sense, it being aimed at users, not developers.

Reply:
LGTM.
FWIW users see annotations as often as developers if they have a good IDE. The docstring and annotated argument should be kept in lock-step.

Reply:
OK, maybe raise an issue about linking the glossary to corresponding typing items?

Reply:
Done!